content/post/deploy-mssql-server-in-aks/index.md

```diff
@@ -0,0 +1,454 @@
+
+```bash
+$ kubectl create namespace demo
+namespace/demo created
+```
+
+
+### Install Cert Manager
+
+To manage TLS certificates within Kubernetes, we need to install [Cert Manager](https://cert-manager.io/). Follow these steps to install Cert Manager using the YAML manifest:
+
+```bash
+$ kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.15.0/cert-manager.yaml
+namespace/cert-manager created
+...
+```
+
+### Create Issuer
+
+Next, we need to create an Issuer, which will be used to generate certificates for TLS settings and internal endpoint authentication of availability group replicas.
+
+Start by generating CA certificates using OpenSSL:
+```bash
+$ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout ./ca.key -out ./ca.crt -subj "/CN=MSSQLServer/O=KubeDB"
+Generating a RSA private key
+..........................................................................+++++
+..+++++
+writing new private key to './ca.key'
+-----
+```
+
+Create a Kubernetes Secret to store the CA certificate:
+```bash
+$ kubectl create secret tls mssqlserver-ca --cert=ca.crt  --key=ca.key --namespace=demo
+secret/mssqlserver-ca created
+```
+
+Create an Issuer using the CA certificate stored in the `mssqlserver-ca` Secret. Below is the YAML definition for the Issuer:
+```yaml
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: mssqlserver-issuer
+  namespace: demo
+spec:
+  ca:
+    secretName: mssqlserver-ca
+```
+
+Create an Issuer using the CA certificate stored in the `mssqlserver-ca` Secret. 
+Below is the YAML definition for the Issuer:
+```bash
+$ kubectl apply -f issuer.yaml
+issuer.cert-manager.io/mssqlserver-issuer created
+```
+
+## Deploy MSSQL Server Availability Group Cluster
+
+Now, we can deploy an MSSQL Server Availability Group Cluster using the following YAML definition:
+
+```yaml
+apiVersion: kubedb.com/v1alpha2
+kind: MSSQLServer
+metadata:
+  name: mssqlserver-ag-cluster
+  namespace: demo
+spec:
+  version: "2022-cu12"
+  replicas: 3
+  topology:
+    mode: AvailabilityGroup
+    availabilityGroup:
+      databases:
+        - music
+  internalAuth:
+    endpointCert:
+      issuerRef:
+        apiGroup: cert-manager.io
+        name: mssqlserver-issuer
+        kind: Issuer
+  tls:
+    issuerRef:
+      name: mssqlserver-issuer
+      kind: Issuer
+      apiGroup: "cert-manager.io"
+    clientTLS: false
+  storageType: Durable
+  storage:
+    storageClassName: "default"
+    accessModes:
+      - ReadWriteOnce
+    resources:
+      requests:
+        storage: 1Gi
+  deletionPolicy: WipeOut
+```
+
+Let's save this yaml configuration into `mssqlserver-ag-cluster.yaml` 
+Then apply the above MSSQL Server yaml,
+
+```bash
+$ kubectl apply -f mssqlserver-ag-cluster.yaml 
+mssqlserver.kubedb.com/mssqlserver-ag-cluster created
+```
+
+In this yaml,
+- `spec.version` is the name of the MSSQLServerVersion CR. Here, a MSSQLServer of `version 2022-cu12` will be created.
```

Comment:
Add: `spec.topology.availabilityGroup.databases` specifies the list of databases that must be added to the `Availability Group` cluster.